freetextresponse/freetextresponse.py

```diff
@@ -599,6 +635,76 @@ def save_reponse(self, data, suffix=''):
         }
         return result
 
+    def store_student_response(self, answer, record_response=True):
+        """
+        Store response from students to the backend
+        """
+        student_id = self.get_student_id()
+        if record_response:
+            self.offer_answer(
+                answer,
+                student_id
+            )
+
+    def offer_answer(self, answer, student_id):
+        """
+        Submit a student answer to the answer pool by appending the given answer to
+        the end of the list.
+
+        Args:
+            pool (dict): answer pool
+                format:
+                [
+                    {
+                        'student_id': ...
+                        'answer': answer
+                    }
+                    {
+                        ...
+                        ...
+                    }
+                    ...
+                ]
+            answer: the answer the student gave
+            student_id: student identifier
+        """
+        # if the answer is wrong, do not display it
+        if self.score != Credit.full.value:
+            return
+
+        # remove any previous answers the student submitted
+        for index, response in enumerate(self.system_answers):
+            if response['student_id'] == student_id:
+                del self.system_answers[index]
+                break
+
+        self.system_answers.append({
+            'student_id': student_id,
+            'answer': answer,
+        })
+
+        # Want to store extra response so student can still see
+        # MAX_RESPONSES answers if their answer is in the pool.
+        if len(self.system_answers) > MAX_RESPONSES + 1:
+            del self.system_answers[0]
```

Comment:
While hopefully unlikely, if two extra answers were ever added to the answer list, we wouldn't ever clean it up, since this is written to trim at most one item, never more.
Instead, if we use a list slice, we can grab the exact subset we want in a single line.

```python
self.system_answers = self.system_answers[-(MAX_RESPONSES+1):]
```

By using a negative value, the slice counts backward from the end of the string in determining where to begin. And since there's nothing after the colon, it grabs until the end of the string.
This behaves no matter the size of the list, even empty.
In our case, it translates to "move back up to 4 items back starting from the end of the list, then grab everything between there and the end."
We don't even need an `if` check!